student/src/main/java/tds/student/sql/repository/ExamRepository.java

```diff
@@ -97,5 +98,23 @@
    * @param examPrintRequest The {@link tds.exam.ExamPrintRequest} containing print data
    * @throws ReturnStatusException
    */
-  void createPrintRequest(ExamPrintRequest examPrintRequest) throws ReturnStatusException;
+  void createPrintRequest(final ExamPrintRequest examPrintRequest) throws ReturnStatusException;
+
+  /**
+   * Creates a segment approval request for an exam segment
+   *
+   * @param examId The id of the exam
+   * @param segmentApprovalRequest A {@link tds.exam.SegmentApprovalRequest} containing request data
+   * @throws ReturnStatusException
+   */
+  void waitForSegmentApproval(final UUID examId, final SegmentApprovalRequest segmentApprovalRequest) throws ReturnStatusException;
+
+  /**
+   * Creates a request to exit a segment
+   *
+   * @param examId The exam id of the {@link tds.exam.ExamSegment} to exit
+   * @param segmentPosition The segment position of the {@link tds.exam.ExamSegment} to update
+   * @throws ReturnStatusException
+   */
+  void exitSegment(final UUID examId, final int segmentPosition) throws ReturnStatusException;
```

Comment:
I think we should probably have an examSegmentRepository that would call exam service.  This is becoming quite the kitchen sink class.